### ISSUE TITLE
@observes doesn't work

### ISSUE THREAD

**Issue by chilicoder** [ember-decorators/ember-decorators#191]
Hi, I'm trying to use observes but it doesn't work

```
export default class IndexController extends Controller {
  one = 'foo';
  @alias('one') two
  three = 'foo';


  @observes('one')
  oneObserver() {
    this.set('three', this.get('one'));
  }

  @observes('two')
  twoObserver() {
    this.set('three', this.get('two'));
  }

  @action
  handleClick() {
    this.set('one', 'bar');
    debug('handleClick');
  }
}
```

When I change `one` property (` this.set('one', 'bar');` ) it doesn't fire `oneObserver` handler.

Can you help?

Here is repo with nonworking code 
https://github.com/chilicoder/ember-decorator-demo2 
https://chilicoder.github.io/ember-decorator-demo2/

**Comment by pzuraq** [comment]
Hi @chilicoder, thanks for opening up this issue, it'll be good to track it here!

This is a known issue that was discussed in the [ES Class RFC](https://github.com/emberjs/rfcs/blob/master/text/0240-es-classes.md#observers-and-events) and will need to be fixed upstream in Ember itself. I'm expecting us to begin fixing it sometime early next year, and we'll definitely post updates once it is done!

In the meantime, I recommend not using observers (they're considered an anti-pattern in Ember in general) or using `Controller.extend` if you still need them.

**Comment by chilicoder** [comment]
Hi @pzuraq, thanks for the answer. I guess @on('didInsertElement') doesn't works because of the same (though didInsertElement hook works nicely)

I rewrote my code to using actions and not to use observer.

**Comment by pzuraq** [comment]
We've removed `@observes` in the v2.0.0-beta.1 release for now, since we're dropping support for object syntax and only supporting ES class syntax. Once changes have been made upstream in Ember we'll attempt to bring them back.

Will close this issue once we get to v2.0.0 stable

**Comment by goroskob** [comment]
2.0.0 was released. Any updates on this?

**Comment by buschtoens** [comment]
> Once changes have been made upstream in Ember we'll attempt to bring them back.

Did not happen yet. 😉

**Comment by pzuraq** [comment]
I think the way I worded that may have been confusing. @chilicoder’s suggestion was to remove @observes and @on from the public API of this library, and when that was completed (upon release of 2.0) I planned on closing this issue.

We do still intend to bring those decorators back, but that will be a much larger effort due to the work required upstream in Ember so I can’t say for sure when that’ll be able to happen. We should probably open another tracking issue for that.

**Comment by pzuraq** [comment]
Closing this in favor of #232
